src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/BurdenEstimateLogic.kt

```diff
@@ -172,26 +169,38 @@ class RepositoriesBurdenEstimateLogic(private val modellingGroupRepository: Mode
         {
             throw UnknownObjectError(touchstoneVersion.id, TouchstoneVersion::class)
         }
-        //Check that scenario exists
+
         scenarioRepository.checkScenarioDescriptionExists(path.scenarioId)
     }
 
-    private fun missingRows(countryMapEntry: Map.Entry<String, HashMap<Short, HashMap<Short, Boolean>>>): Boolean
+    private fun rowErrorMessage(countriesWithMissingRows: Map<String, HashMap<Short, HashMap<Short, Boolean>>>): String
     {
-        return countryMapEntry.value.any { a ->
-            a.value.any { y -> !y.value }
-        }
+        val countries = countriesWithMissingRows.keys
```

Comment:
In general, the whole way the logic code around missing rows deals with Map<String, HashMap<Short, HashMap<Short, Boolean>>> and other such opaque structures is pretty confusing. Laying the code out like this with clear val names, does really help understanding, though as you'll see I still managed to get confused about the test! It would probably be even clearer to write a class or two with obvious property/method names rather than using lengthy generics - up to you whether you think it's worth refactoring though.

Reply:
Agreed. I've added some type aliases and extension methods which I hope makes things a little clearer at the point of use.